### DISCUSSION TITLE
TP5 - ejercicio 3 - ¿qué situaciones de error hay?

### DISCUSSION THREAD

**Discussion by matiaspalleres** [INGCOM-UNRN-P1/cursada-2023#164] [Preguntas y respuestas] [answered]
Buenas tardes, estoy haciendo el punto 3 del tp5 y no puedo hacer las situaciones de error como las venia haciendo con los otros tp, porque el programa no compila cuando asigno un retorno de tipo estructura a una constante de tipo entero. ¿Como podria solucionar esto?
Edit: Particularmente en la funcion suma_de_fracciones
Dejo el link al repositorio:  https://github.com/INGCOM-UNRN-P1/tp5-2023-matiaspalleres

**Comment by martinvilu** [comment, marked as answer]
```C
fraccion_t suma_de_fracciones (fraccion_t fraccion_1, fraccion_t fraccion_2, fraccion_t fraccion_resultado)
```
Los argumentos de tipo “estructura”, funcionan igual que los argumentos simples que vimos al principio; `int`, `char`, `double`, etc.

Y siendo que en C todo pasa por copia, en el tercer argumento no estaría funcionando como ‘retorno’, ya que solo cambia la ‘copia’.

Por otro lado, también hay menos errores posibles sin usar punteros.

**Reply by martinvilu** [reply]
El problema con
```C
retorno = DIVISION_POR_CERO;
```
Es que lo que retornas, tiene que ser del mismo tipo, que en este caso es un `fraccion_t`, no un `int`.

Al sumar, ¿la división por cero te impide calcular? (lo mismo al mostrar, que la fracción sea 0, solo impide que lo 'transformes' en un número real (`double`/`float`))
